src/main/fc/rc_adjustments.c

```diff
@@ -237,7 +242,19 @@ static const adjustmentConfig_t defaultAdjustmentConfigs[ADJUSTMENT_FUNCTION_COU
         .adjustmentFunction = ADJUSTMENT_YAW_F,
         .mode = ADJUSTMENT_MODE_STEP,
         .data = { .step = 1 }
+    },
+#ifdef USE_OSD_PROFILES
```

Comment:
The elements in `defaultAdjustmentConfigs` cannot be conditionally assigned. The Configurator references the index and is unaware whether the connected firmware has the code compiled in or not. So if additional elements are added then the index position referenced by the configurator will be incorrect. So even if the supporting code is defined out, the index for this entry must be defined and permanent.

Reply:
`defaultAdjustmentConfigs` is same length as `adjustmentFunction_e` (ADJUSTMENT_FUNCTION_COUNT). Using designated initializer will place ADJUSTMENT_OSD_PROFILE to correct index or keep the entry zeroed when USE_OSD_PROFILES is not defined.
Configurator should use `adjustmentFunction_e`, which is before this PR one to one mapped with defaultAdjustmentConfigs

Reply:
@ledvinap While at one point `defaultAdjustmentConfigs` and `adjustmentFunction_e` were synchronized, this was changed some time back to no longer be the case. The configurator (and the `adjrange` cli command index) reference the entry in `defaultAdjustmentConfigs` and as such there is no requirement that they match the indexes of `adjustmentFunction_e`. In fact this opens up opportunities (currently not used) where the same adjustment function could be exposed with different methods of adjustment - like with a "step" entry and a "selection" entry. So yes it's legacy code that the length of `defaultAdjustmentConfigs` is still set to ADJUSTMENT_FUNCTION_COUNT. It should probably have it's own defined length that matches the actual entries configured.

Reply:
@etracer65 : 
Yes, you are right.

Using array index as public value shared over CLI and MSP is problematic, some refactoring would be nice - using new enum to index `defaultAdjustmentConfigs` and eventually using this to lookup in the array.
BTW: that means that for example  ADJUSTMENT_ROLL_RC_RATE can't be used?

Reply:
Yes, rc adjustments need a complete refactoring as it's a convoluted mess. But that's obviously outside the scope of this PR ;)

And yes, `ADJUSTMENT_ROLL_RC_RATE` is not implemented yet. Same with `ADJUSTMENT_PITCH_RC_RATE`, `ADJUSTMENT_ROLL_RC_EXPO`, and `ADJUSTMENT_PITCH_RC_EXPO`.  These are currently not exposed in the Configurator. Partly the logic is that the individual roll/pitch rc_rate and expo values also aren't exposed in the GUI and can only be set in CLI. These are pending items for Configurator updates. So in conjunction with adding the adjustments to the Configurator they'll need to be added to the end of `defaultAdjustmentConfigs`.

Reply:
@etracer65 I think I'm starting to understand now.  I did tie the index into defaultAdjustmentConfigs  to adjustmentFunction_e.  I'll rebase, apply the comments below and test again.  I have to agree with ledvinap, this will cause bugs in the future.  Aligning with adjustmentFunction_e as the source of truth not only makes sense, it forces the Configurator to use the correct value somewhat and also makes it easy/clear to know what value to use in the CLI with the adjrange command.  Anyway it is what it is - refactor for later...

Reply:
@etracer65 I think I'm starting to understand now.  I did tie the index into defaultAdjustmentConfigs  to adjustmentFunction_e.  I'll rebase, apply the comments below and test again.  I have to agree with ledvinap, this will cause bugs in the future.  Aligning with adjustmentFunction_e as the source of truth not only makes sense, it forces the Configurator to use the correct value somewhat and also makes it easy/clear to know what value to use in the CLI with the adjrange command.  Anyway it is what it is - refactor for later...